### DISCUSSION TITLE
Add playfield anchor points for mania

### DISCUSSION THREAD

**Discussion by alexpado** [ppy/osu#24379] [Ideas] [answered]
*The issue has been sent in #22890 but the discussion title doesn't really fit the issue and was for another issue in the first place.*

While skining mania, users may want to "attach" some skin element to the playfield. It works well using current anchors point until... you change the key count.

If you configure your skin in a 4K setup, when switching to a 7K setup, every skin element you tried to anchor to your playfield will overlap the playfield.

Example:

**4K Setup**
![image](https://github.com/ppy/osu/assets/31733699/bac8af55-8700-4f49-92d8-4d40a25ce6f2)

**7K Setup (unmodified)**
![image](https://github.com/ppy/osu/assets/31733699/49760f3b-8245-411d-a8b2-4be772704feb)

As you can see, all elements selected in the 4K setup are overlapping in 7K. The idea would be to add these anchors points for mania:

- Field Top Left
- Field Top Centre
- Field Top Right
- Field Centre Left
- Field Centre (which *should* be the current `Centre` one, if the playfield is always centered)
- Field Centre Right
- Field Bottom Left
- Field Bottom Centre
- Field Bottom Right

In my situation, I would use the `Field Centre Right` which could snap my elements to the playfield, moving around depending on the playfield width.

*As a bonus, it would be cool to have a togglable grid when editing skin, to align elements*

**Comment by peppy** [comment, marked as answer]
Both of these ideas are already discussed and confirmed for implementation at some point.

**Reply by peppy** [reply]
https://github.com/ppy/osu/pull/24387

**Reply by alexpado** [reply]
My skin works perfectly with the linked PR 👍🏼
